arangod/Aql/IResearchViewOptimizerRules.cpp

```diff
@@ -305,74 +296,168 @@ bool optimizeScoreSort(IResearchViewNode& viewNode, ExecutionPlan* plan) {
   // we've found all we need
   auto const& sortElements = sortNode->elements();
   TRI_ASSERT(!sortElements.empty());
-  std::vector<std::pair<size_t, bool>> scoresSort;
-  auto zero = std::numeric_limits<size_t>::max();
+  std::vector<HeapSortElement> heapSort;
+  std::vector<std::vector<latematerialized::ColumnVariant<true>>> usedColumns;
+  auto const& primarySort = getPrimarySort(viewNode.meta(), viewNode.view());
+  auto const& storedValues = getStoredValues(viewNode.meta(), viewNode.view());
+  auto const columnsCount = storedValues.columns().size() + 1;
+  usedColumns.resize(columnsCount);
+  std::vector<
+      latematerialized::AttributeAndField<latematerialized::IndexFieldData>>
+      attrs;
+  // maps attribute bucket to actual sort bucket
+  containers::FlatHashMap<size_t, size_t> storedMaps;
   for (auto const& sort : sortElements) {
     TRI_ASSERT(sort.var);
-
     auto const* varSetBy = plan->getVarSetBy(sort.var->id);
     TRI_ASSERT(varSetBy);
-
-    aql::Variable const* sortVariable{};
     switch (varSetBy->getType()) {
+      case ExecutionNode::ENUMERATE_IRESEARCH_VIEW: {
+        if (&viewNode != varSetBy) {
+          // we can't use column from another index
+          return false;
+        }
+        auto source = viewNode.getSourceColumnInfo(sort.var->id);
+        TRI_ASSERT(source.first != std::numeric_limits<ptrdiff_t>::max());
+        heapSort.push_back(HeapSortElement{.postfix = "",
```

Comment:
```suggestion
        heapSort.push_back(HeapSortElement{
```

Reply:
It was like that from the very beginning. But for some reason it failed to compile on our CI with warning "member not initialized".

Reply:
Now it should be ok, we merged PR, to remove this useless warning, could you try again please?

Reply:
I'll give it a try when finish with tests fixes